src/main/scala/io/smartdatalake/workflow/dataobject/DeltaLakeTableDataObject.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2020 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.smartdatalake.workflow.dataobject
+
+import io.smartdatalake.config.SdlConfigObject.{ConnectionId, DataObjectId}
+import io.smartdatalake.config.{ConfigurationException, FromConfigFactory, InstanceRegistry}
+import io.smartdatalake.definitions.DateColumnType
+import io.smartdatalake.definitions.DateColumnType.DateColumnType
+import io.smartdatalake.util.hdfs.{HdfsUtil, PartitionValues}
+import io.smartdatalake.util.hive.HiveUtil
+import io.smartdatalake.util.misc.{AclDef, AclUtil}
+import io.smartdatalake.workflow.connection.HiveTableConnection
+import org.apache.hadoop.fs.FileSystem
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, Row, SaveMode, SparkSession}
+
+import scala.collection.JavaConverters._
+
+case class DeltaLakeTableDataObject(override val id: DataObjectId,
```

Comment:
Can you add scaladoc for attributes here? you can check other data objects if there are already for these attributes and copy/paste

Reply:
Done: https://github.com/smart-data-lake/smart-data-lake/pull/13/commits/7c147455d1d452fd9b69d429fbd18e54b73c4da7